lib/ddtrace/workers.rb

```diff
@@ -12,6 +12,7 @@ class AsyncTransport
       DEFAULT_TIMEOUT = 5
       BACK_OFF_RATIO = 1.2
       BACK_OFF_MAX = 5
+      SHUTDOWN_TIMEOUT = 1
```

Comment:
I noticed this went from 5 to 1. What is the reason for it? Shutdown timeout and back off max used to be the same, they are not any more, just wondering if there are some side effets.

Reply:
Yep, that's something worth discussing. I still have to try out different values before merging it, but these are a few points to consider:

* The backoff duration doesn't have to meet (or relate somehow) to the shutdown timeout anymore, because we wake up the thread immediately when we trigger the shutdown process;

* I have the impression that 5 seconds was just an initial guess (maybe @ryplo or @palazzem can elaborate on that), but it has proven to be way too long. The thing is, we only care about the shutdown process for "short-lived processes" like `resque` workers, and a lot of times `resque` jobs are in the order of a couple miliseconds, so if anything goes wrong and we actually wait for the timeout we would impact greatly the throughput of a `resque` worker pool. I think that in that case would be preferably to lose data than impact the performance of the application.

* Having said all that, the important thing here is to set up a reasonable hard limit for "one last flush". Do you guys think 1 second is a reasonable amount of time? I guess if we had a very long trace (like originating from a 20 minute `resque` job) that could be a problem, but on the other hand #247 should mitigate this soon.

Reply:
Yes, if I recall correctly when we wrote the `shutdown!` method we guessed the `SHUTDOWN_TIMEOUT` based on the `DEFAULT_TIMEOUT`. I think based on some testing that I've done with a sample app and our integration that 1 second should be reasonable.

Reply:
Thanks @p-lambert for the explanation. Sounds legit and 1 sec is fine IMHO.

Reply:
totally agree, 1 second is fine since it's not used anymore by the Resque integration so we don't have any impact here.